freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -74,33 +77,39 @@
+    private InstanceStatus updateStatuses(CloudVmInstanceStatus vmInstanceStatus, InstanceMetaData instanceMetaData,
+        Map<InstanceMetaData, DetailedStackStatus> instanceHealthStatusMap) {
         LOGGER.info(":::Auto sync::: {} instance metadata status update in progress, new status: {}",
                 instanceMetaData.getShortHostname(), vmInstanceStatus);
         InstanceStatus status = null;
-        switch (vmInstanceStatus.getStatus()) {
-            case STARTED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.CREATED);
-                status = InstanceStatus.CREATED;
-                break;
-            case STOPPED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.STOPPED);
-                status = InstanceStatus.STOPPED;
-                break;
-            case FAILED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.FAILED);
-                status = InstanceStatus.FAILED;
-                break;
-            case TERMINATED:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.DELETED_ON_PROVIDER_SIDE);
-                status = InstanceStatus.DELETED_ON_PROVIDER_SIDE;
-                break;
-            case TERMINATED_BY_PROVIDER:
-                setStatusIfNotTheSame(instanceMetaData, InstanceStatus.DELETED_BY_PROVIDER);
-                status = InstanceStatus.DELETED_BY_PROVIDER;
-                break;
-            default:
-                LOGGER.info(":::Auto sync::: the '{}' status is not converted", vmInstanceStatus.getStatus());
+        if (DetailedStackStatus.UNREACHABLE == instanceHealthStatusMap.get(instanceMetaData)) {
+            setStatusIfNotTheSame(instanceMetaData, InstanceStatus.UNREACHABLE);
+            status = InstanceStatus.UNREACHABLE;
+        } else {
+            switch (vmInstanceStatus.getStatus()) {
+                case STARTED:
+                    setStatusIfNotTheSame(instanceMetaData, InstanceStatus.CREATED);
+                    status = InstanceStatus.CREATED;
+                    break;
+                case STOPPED:
+                    setStatusIfNotTheSame(instanceMetaData, InstanceStatus.STOPPED);
+                    status = InstanceStatus.STOPPED;
+                    break;
+                case FAILED:
+                    setStatusIfNotTheSame(instanceMetaData, InstanceStatus.FAILED);
+                    status = InstanceStatus.FAILED;
+                    break;
+                case TERMINATED:
+                    setStatusIfNotTheSame(instanceMetaData, InstanceStatus.DELETED_ON_PROVIDER_SIDE);
+                    status = InstanceStatus.DELETED_ON_PROVIDER_SIDE;
+                    break;
+                case TERMINATED_BY_PROVIDER:
+                    setStatusIfNotTheSame(instanceMetaData, InstanceStatus.DELETED_BY_PROVIDER);
+                    status = InstanceStatus.DELETED_BY_PROVIDER;
+                    break;
```

Comment:
Since the unreachable's logic above is the highest precedence, it appears that unhealthy overrides stopped, failed, terminated, and terminated by the provider (all of which will report an unreachable status). I think the instance's health status should only be considered if the VM is in the running state.

Reply:
You are absolutely right. I made the change so it will not overwrite the failed and terminated status. Now only running instances that are unreachable or unhealthy are reported as it is. 

Tested again below,

<img width="308" alt="Screen Shot 2020-10-26 at 9 59 25 AM" src="https://user-images.githubusercontent.com/67844271/97188858-fc812f00-1771-11eb-895d-172aebb42518.png">
<img width="332" alt="Screen Shot 2020-10-26 at 9 59 14 AM" src="https://user-images.githubusercontent.com/67844271/97188863-fdb25c00-1771-11eb-8ec1-394116747bc9.png">
<img width="315" alt="Screen Shot 2020-10-26 at 9 59 09 AM" src="https://user-images.githubusercontent.com/67844271/97188872-0014b600-1772-11eb-8b42-8de6d7c2c0dc.png">